packages/fxa-settings/src/pages/ConnectAnotherDevice/index.tsx

```diff
@@ -0,0 +1,249 @@
+  isSignedIn: boolean; // check if `isSignedInAccount` (on "account"), or get `isVerificationSameBrowser` on the broker
+  showSuccessMessage?: boolean; // check if you have a truthy value for "showSuccessMessage" either on the model, or in a search param by the same name. This is so that if you hit this url, you don't automatically be an inaccurate status message
+  isSignUp: boolean; // we get this by getting the verification reason (or "type") off of the model
+  isSignIn: boolean; // we get this by getting the verification reason (or "type") off of the model
+  canSignIn: boolean; //we check if the user is 1. not signed in, and 2. on a UA that supports sync auth (i.e., Firefox Desktop where version > 40, or Firefox android where version > 43)
+  device?: Devices;
+};
+
+export enum Devices {
+  FIREFOX_MENU = 'Firefox Menu',
+  FIREFOX_ANDROID = 'Firefox Android',
+  FIREFOX_DESKTOP = 'Firefox Desktop',
+  FIREFOX_IOS = 'Firefox Ios',
+  OTHER_ANDROID = 'Other Android',
+  OTHER_IOS = 'Other Ios',
+  OTHER = 'Other',
+}
+export const viewName = 'connect-another-device';
+const ConnectAnotherDevice = ({
+  email,
+  forceView = false,
+  entrypoint,
+  isSignedIn,
+  showSuccessMessage = false,
+  isSignUp,
+  isSignIn,
+  canSignIn,
+  device = Devices.FIREFOX_MENU,
+}: ConnectAnotherDeviceProps & RouteComponentProps) => {
+  usePageViewEvent(viewName, REACT_ENTRYPOINT);
+  const getEscapedSignInUrl = () => {
+    // TODO:
+    // recreate the escaped signin url.
+    // basically this means pulling a bunch of params off of the relier
+    // you can see this in the `sync-auth-mixin.js`, and in `connect_another_device.js`
+    return '/signin';
+  };
+
+  const isEligibleForPairing = () => {
+    // TODO:
+    // get context from `relier.get('context')`
+    // originally we got the entrypoint from the relier, but here we have it as a prop
+    // get the action from the relier as well
+    // if:
+    //    the user is signed into the default account,
+    //    AND the context === FX_DESKTO_V3_CONTEXT
+    //    AND (
+    //       EITHER the entrypoint === FIREFOX_TOOLBAR_ENTRYPOINT)
+    //       OR (the action !== email, and the entrypoint is firefox preferences || firefox synced tabs || firefox tabs sidebar || firefox menu || firefox fx view )
+    // then:
+    const isEligible = false;
+    if (isEligible) {
+      return true;
+    }
+    return false;
+  };
+  useEffect(() => {
+    if (!forceView && isEligibleForPairing()) {
+      // To avoid redirection loops the forceView property might be set
+      navigate('/pair');
+    }
+  }, [forceView]);
+
+  return (
+    <div>
+      {showSuccessMessage && (
+        <>
+          {isSignedIn && (
+            <FtlMsg
+              id="connect-another-device-signed-in-header"
+              elems={{
+                subheader: (
```

Comment:
We actually aren't going to be using the css class -- I realized it would be better/more accessible to use the svg inline -- but I think your solution will still work!! I'll try it out!

Reply:
Fixed!! This worked!